### DISCUSSION TITLE
Please add a json selector to the UI - and please remove the XL from the name

### DISCUSSION THREAD

**Discussion by OlivioSarikas** [MoonRide303/Fooocus-MRE#121] [Ideas]
Please add a json selector to the UI, so we can load our custom lists on the fly.

Please remove the XL from the name, as this also works well with SD 1.5 and a lot of people don't or can't use XL, so it is more difficult to make videos or posts about it and it is confusing the community of users

**Comment by barepixels** [comment]
Check out Olivio Sarikas video on Style Selector

**[Style Selector for A1111 + ChatGPT HACK!!!!](https://youtu.be/6aZONG75dr0?si=UmIXYbze1hW92g1V)**

**Comment by 3Diva** [comment]
@OlivioSarikas Did you intend to leave that comment under the Fooocus-MRE discussion? Or did you intend to leave it under the Style Selector XL github?

**Comment by barepixels** [comment]


**Reply by 3Diva** [reply]
@barepixels You can already add different style json files to Fooocus-MRE.

**Reply by barepixels** [reply]
nooooo  I mean in one style sheet, to be able to drag a specific style such as "Artstyle Pop Art" and move it to the 3rd position from the top.  Basically, drag and drop your favorite ones to the top.  Reordering then export

**Reply by MoonRide303** [reply]
@barepixels Styles are being listed in following order:
1) default filenames (if they're present): `default_styles_files = ['sdxl_styles_sai.json', 'sdxl_styles_twri.json', 'sdxl_styles_diva.json', 'sdxl_styles_mre.json']`,
2) any other `.json` file, alphabetically.

So if you want order change, you can simply use names like "styles_1.json", "styles_2.json", etc. You can do that for sdxl*.json styles, too - just the "Default (Slightly Cinematic)" in coming from the code. So for example if I do something like that:
![image](https://github.com/MoonRide303/Fooocus-MRE/assets/130458190/152aa52c-a208-4757-bae2-770e937d6256)
then It will be ordered accordingly in the UI:
![image](https://github.com/MoonRide303/Fooocus-MRE/assets/130458190/01aadef7-6f9b-4681-b1b9-493a77339fc7)

Re-ordering json content is something you can do in any text editor. And if you'd like to do it in a smarter way you can use tools like [jq](https://jqlang.github.io/jq/). For example if you wanted to have all your style sorted alphabetically by name, you can do something like that:
`type styles-in.json | jq "sort_by(.name)" > styles-out.json` (Windows) or
`cat styles-in.json | jq "sort_by(.name)" > styles-out.json` (*nix).

**Reply by barepixels** [reply]
@MoonRide303 Found a solution

For example, change ones I like to:
{
"name": "Pop Art",
"prompt": "Pop Art style {prompt} . Bright colors, bold outlines, popular culture themes, ironic or kitsch",
"negative_prompt": "ugly, deformed, noisy, blurry, low contrast, realism, photorealistic, minimalist"
},

to:
{
"name": "001_Pop Art",
"prompt": "Pop Art style {prompt} . Bright colors, bold outlines, popular culture themes, ironic or kitsch",
"negative_prompt": "ugly, deformed, noisy, blurry, low contrast, realism, photorealistic, minimalist"
},

then sorted alphabetically by "name"

No coding needed

**Comment by barepixels** [comment]
@OlivioSarikas I just tested, while 1.5 models seem to work, even with refiner.  1.5 loras does not.  Here is what @MoonRide303 said about support for 1.5 or 2.1  https://github.com/MoonRide303/Fooocus-MRE/issues/29

EDIT:  Am thinking 1.5 model does not work... I think it doesn't show errors but defaults to SDXL base

For example, I tried to recreate this in Fooocus and results looks similar to SDXL base NOT Deliberate
https://civitai.com/images/183148?modelVersionId=15236&prioritizedUserIds=4055&period=AllTime&sort=Most+Reactions&limit=20

**Comment by MoonRide303** [comment]
@OlivioSarikas You can add files with any names you want on style files you add into sdxl_styles folder - any file ending with `.json` will be loaded (as long as it has compatible syntax). Default file names starting with "sdxl" are result of two things:
1) those were the original names used in the repository they originally came from: [sdxl_prompt_styler](https://github.com/twri/sdxl_prompt_styler),
2) while styles using commonly recognized phrases can work pretty well between different models (and generations), it's not so obvious they will work as expected - different models might be trained using different keywords, so it's best to not make too far-reaching assumptions about styles compatibilities. Styles that were created and tested with SDXL model should be marked as sdxl, IMO.

Non-SDXL models are not currently supported - trying to load anything non-SDXL it will just result in reverting to the default model. As I previously mentioned and (in the issue mentioned by @barepixels) it is theoretically possible (even hybrid SD base + SDXL refiner workflows using VAE decode/encode at switch step), but it would complicate the flow - normalisation to model-native resolutions where it's needed, inpainting model being delivered as patch for SDXL architecture, LoRAs compatibility etc.

I could replace protection from loading non-SDXL models and replace it with warning messege, though - so people who know what they're doing could play with previous generation (it should work fine after disabling refiner, and setting resolution to something around 512x512).

**Comment by barepixels** [comment]
protection from loading non-SDXL models  meaning models and  loras?

we can add a set of 1.5 and 2.1 in our resolutions.json :)

Ideally, if choose 1.5 or 2.1 refiner should be off and hidden

With that said, I rather see you get DW Pose working before supporting 1.5 and 2.1  haha

**[DW Pose for A1111 - Better than OpenPose ControlNET](https://youtu.be/30b2k1p2CiE?si=3cG8Whk0VCngyJO4)**

on YouTube I saw someone wrote in the comment 
For those interested, the DWPose preprocessor also works great for **ComfyUI** and ControlNet

**Reply by MoonRide303** [reply]
@barepixels You can check out v2.0.78.3, I've just added limited support for SD models. Functions that will not work with non-SDXL models: refiner, Control-LoRAs, Revision, inpainting, outpainting.

**Reply by barepixels** [reply]
Oh yes.  Deliberate v3, SD1.5, Prompt: redhead

![2023-09-27_07-13-15_3108](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/55b7e787-4a13-44d6-90d6-afcf8f400a6b)

Total time: 8.39 seconds on 3070 8GB  512 x 768

**Reply by barepixels** [reply]
Switch to SDXLbase and Refiner > Upscale Fast 2x

![2023-09-27_09-32-03_5891](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/7a520b90-b7a4-428a-913f-e8f465dc9407)

Total time: 10.80 seconds 1024 x 1536, **NO METADATA**

**Reply by MoonRide303** [reply]
@barepixels Fast upscale fixed (another side effect of merging changes from vanilla - MRE flow wasn't compatible with "direct_return" from vanilla). Restart to pull update, should be fine now.

**Reply by barepixels** [reply]
update Total time: 10.80 seconds, **NO METADATA**

**Comment by barepixels** [comment]
Switch to SDXLbase and Refiner > img2img scale 2
![2023-09-27_07-42-18_2213](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/bfd3bc24-8b0f-4306-9ab1-d1bd298d2dff)
Total time: 48.58 seconds  1024 x 1536, METADATA included

**Comment by barepixels** [comment]
Switch to SDXLbase and Refiner > Upscale 2x  (NOT FAST)  Judging by her eyes, this is best out of 3 from Fooocus
![2023-09-27_07-48-19_8398](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/8df037ca-bab9-4b1c-99e6-4406e9ba19e0)
Total time: 38.59 seconds  1024 x 1536, METADATA included

**Comment by barepixels** [comment]
Topaz GigaPixels (standard-scale-2_00x)

![2023-09-27_07-22-51_3549-standard-scale-2_00x](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/0cbbdcea-f86c-4628-bfa9-c99c44e3c4d4)
Total Time: N/A but very fast  1024 x 1536, METADATA included, **UPDATE: Foooocus can load prompt from Topaz GigaPixels output**

**Reply by MoonRide303** [reply]
@barepixels Fast upscale doesn't involve diffusion model - it's just simple ESRGAN type upscaler (not using prompt or any other generation parameters), so saving metadata doesn't make much sense in this case.

**Reply by barepixels** [reply]
understood, base on test images above, I won't be using Fast 2x anyway.
